sdk/core/core/src/az_span.c

```diff
@@ -618,3 +618,56 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index)
   }
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
+
+AZ_NODISCARD az_span _az_span_trim_white_space(az_span source)
+{
+  // Trim from end after trim from start
+  return _az_span_trim_white_space_from_end(_az_span_trim_white_space_from_start(source));
+}
+
+AZ_NODISCARD AZ_INLINE bool _az_is_white_space(uint8_t c)
+{
+  switch (c)
+  {
+    case ' ':
+    case '\t':
+    case '\n':
+    case '\r':
+      return true;
+  }
+  return false;
+}
+
+AZ_NODISCARD az_span _az_span_trim_white_space_from_start(az_span source)
+{
+  // loop from start to the first non white space
+  uint8_t* source_ptr = az_span_ptr(source);
+  for (int32_t index = 0; index < az_span_size(source); ++index)
+  {
+    if (!_az_is_white_space(source_ptr[index]))
+    {
+      return az_span_slice_to_end(source, index);
+    }
+  }
+  // reaching here means it was all white spaces
+  // Return az_span of size 0, using AZ_SPAN_NULL for now so we dont call slice
+  return AZ_SPAN_NULL;
+}
+
+AZ_NODISCARD az_span _az_span_trim_white_space_from_end(az_span source)
+{
+  // loop from end to the first non white space or 0
+  uint8_t* source_ptr = az_span_ptr(source);
+  for (int32_t index = az_span_size(source); index > 0;)
+  {
+    --index;
+    if (!_az_is_white_space(source_ptr[index]))
+    {
+      // slice from start to index plus 1 as slice end_index is inclusive
+      return az_span_slice(source, 0, index + 1);
```

Comment:
We can avoid this `index + 1` if we start `index` of by `az_span_size(source) - 1` and move the `--index` into the for-loop (i.e. do it at the end).